### PR TITLE
Changed BoundedUserIdentity from a map to struct

### PR DESCRIPTION
**Pull request by samiulsami**

### CODE REVIEW

File: james/types.go [hunk 1]
```diff
@@ -238,6 +238,12 @@ func (o ObjectIdentifier) AdditionalTokenClaims() *jwt.MapClaims {
 	return o.AdditionalClaims
 }
 
-func (o ObjectIdentifier) GetBoundedUserIdentity() map[string]string {
+func (o ObjectIdentifier) GetBoundedUserIdentity() BoundedUserIdentityType {
 	return o.BoundedUserIdentity
 }
+
+type BoundedUserIdentityType struct {
```
Comment (Imtiaz246) [inline james/types.go]: Just name it as `UserIdentity`

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: james/types.go (hunk 1 above)
Comment (samiulsami) [reply james/types.go]: fixed

File: james/helpers.go [hunk 2]
```diff
@@ -17,8 +17,15 @@ func generateObjectAddr(object Object) (string, error) {
 		addr = fmt.Sprintf("%s.x&type-acc", addr)
 	}
 
-	for key, value := range object.GetBoundedUserIdentity() {
-		addr = fmt.Sprintf("%s&%s-%s", addr, key, value)
+	bui := object.GetBoundedUserIdentity()
+	if bui.OwnerName != "" {
```
Comment (Imtiaz246) [inline james/helpers.go]: name thsese shorter & symbolic like: `ownername` -> `_ownr_nm`
`ownerid` -> `_ownr_id`
`ownertype` -> _`ownr_typ`

or only shorted: 
`nm`, `id`, `typ`

and write a method for this portion.

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

**Review by Imtiaz246 — APPROVED** [review]
LGTM

**Pull request merged**